src/fetch-papers.js

```diff
@@ -47,6 +47,50 @@ const {
   foldersToPath,
 } = main(dropboxPaperApi);
 
+const appendDocContent = docs => Promise.all(
+  docs.map(doc =>
+    promiseAllProps({
+      ...doc,
+      content: fetchDocContent(doc.id),
+    })
+      .then(doc => ({
+        ...doc,
+        location: path.join(
+          doc.directory,
+          `${kebabCaseIt(doc.content.metaData.title)}.md`
+        ),
+      }))
+  )
+);
+
+const appendMetaData = docs => Promise.all(
+  docs.map(([ id, folders ]) => promiseAllProps({
+    id,
+    folders,
+    directory: path.join('docs', foldersToPath(folders.slice(1))),
+    metaData: fetchDocMetaData(id),
+  }))
+);
+
+const saveDocsLocally = docs => docs.forEach(doc => {
+  mkdir(doc.directory, { recursive: true })
+    .then(() => writeFile(
+      doc.location,
+      `${jsonToFrontmatter(doc.metaData)}${doc.content.body}`
+    ));
+
+  writeFile('docs/meta-tree.json', JSON.stringify(
+    docs.map(({ id, folders, content: { metaData }, location }) => ({
```

Comment:
This map is extra to filter and save only the needed content, should make the file a bit smaller and faster to save.